tests/components/mqtt/test_init.py

```diff
@@ -65,39 +98,32 @@ def sensor_platforms_only():
 
 
 @pytest.fixture(autouse=True)
-def mock_storage(hass_storage):
+def mock_storage(hass_storage: Generator[dict[str, Any], None, None]) -> None:
```

Comment:
The `hass_storage` fixture returns a dict when injected by pytest. It's only a generator before injection.

Reply:
Thnx,  will arrange a follow up PR to correct this!

Reply:
I missed that one...
Currently hass_storage is the only "global" fixture that I have left out of my script because when I looked into it I realised that it might return a list, not necessarily a dict.
It also goes through JSON serialization, so I wondered if maybe it should be JSONValueType/JSONObjectType.

Reply:
Opened https://github.com/home-assistant/core/pull/87986

Reply:
`data` must be a dict:

https://github.com/home-assistant/core/blob/3f910b49ae5296610a4144b9b4e95a31e4a26f3b/tests/common.py#L1207-L1264

Reply:
Yes I realised that when I re-read the function just now.

The mock storage has to be a `dict`. It's the stored data that is bound to `list` or `dict`.